lib/mix/tasks/compile.nerves_hub.ex

```diff
@@ -0,0 +1,19 @@
+defmodule Mix.Tasks.Compile.NervesHub do
+  use Mix.Task
+  @recursive true
+
+  def run(_args) do
+    current_keys = NervesHub.public_keys()
+
+    new_keys =
+      Application.get_env(:nerves_hub, :public_keys, [])
```

Comment:
I'd still like to remove the compiler since it seems like it's only an initialization step. It also overwrites a file in `priv` which feels wrong to me. I suppose it would feel less wrong if there were a big "DO NOT MODIFY THIS FILE SINCE IT IS AUTOGENERATED" at the top of it, but maybe that's not needed at all. How about the configuration option be `:public_keys_file` and require the user to set it if they want to override the default?

Reply:
The file is just storage mechanism and shouldn't be very interesting, which is why I didn't create any read / write functions. The entry point developer should use is the `public_keys` list. This list can contain both string keys and atom names. The atom names need to be looked up through NervesHubCLI during compilation. In addition, the main mix project has the capability of configuring the NervesHub `org_name`. Currently, the CLI expands these values when the `:nerves_hub` dependency gets compiled. When the CLI checks the `Mix.Project.config()[:nerves_hub]` for n `org_name`, it receives the project config from the mix file of `:nerves_hub` instead of the parent app. We could fix this through compilers, aliases, or by requiring a module in the parent project to define a `use` for code expansion.

Reply:
How about none of the above? (i.e., no compilers, no aliases, and no macros). This is a workaround from a `mix` bug, so is it possible to fix `mix`?

Reply:
we can do something like 
```
Module.register_attribute(:external_file, :accumulate)
for f <- Mix.Project.config_files() do
   @external_file f
end
```

I can't remember if this will work or not, but this should cause the file to be recompiled every time a `config.exs` file is changed.